ai/docs/Vicuna Model Deployment and Training Documentation.md

```diff
@@ -0,0 +1,53 @@
+# Vicuna Model Deployment and Training Documentation
+## Introduction
+
+[Vicuna model](https://lmsys.org/blog/2023-03-30-vicuna/) is an open-source chatbot model trained by fine-tuning LLaMA on user-shared conversations collected from ShareGPT. It has superior performance compared to LLaMA and Alpaca, even approaching the performance of ChatGPT, evaluted by GPT-4 and with comparably low training costs. The code and model parameters [in open access with github](https://github.com/lm-sys/FastChat).
+
+This document will focus on deploying and fine-tuning Vicuna model. We re-encapsulate the FastChat API for Vicuna, make it easier for personal use.
+
+Before use, you should have **GPU** in your system, with **enough GPU memory****(14GB for Vicuna-7B and 28GB for Vicuna-13B)**. Also, make sure that **docker, docker-compose, jq** are installed in your Linux system. 
+
+## Deployment
+
+For Deployment of Vicuna, you should first change the directory:
+
+```plain
+cd ./antrea/ai/vicuna/deploy
+```
+We use Dockerfile to build docker container, you should have root privilege to run docker commands.
+### 1.Build container with Vicuna
+
+The environment variables should be set for build Vicuna container, so you should first set the environment variables MODEL_PATH and MODEL_NAMES in "1_build.sh". The MODEL_PATH should be accurate, and the MODEL_NAMES can be randomly named by your preference.
+
+```plain
+export MODEL_PATH=<your_vicuna_model_path>
+export MODEL_NAMES=<random_name> 
+```
+Then run the command:
+```plain
+sudo sh 1_build.sh
+```
+Since the docker-compose command in "1_build.sh", "sudo" is needed. This Shell scripts will first set the environment variables and then run the docker-compose to build container according to "docker-compose.yml" file, it needs 1-2 minutes or so. After container created successfully, you can send request to the url: [http://0.0.0.0:8000/v1/chat/completions](http://0.0.0.0:8000/v1/chat/completions)
+### 2.Check if the model can repsonse
+
+```plain
+sh 2_call.sh
+```
+It will send POST request to call the model, and return the model reponses. If the responses are like this, the model has been deployed successfully. 
+```plain
+{"id":"chatcmpl-UFWP9BqDg3s7DDfc5eXLS3","object":"chat.completion","created":1701329914,"model":"vicuna","choices":[{"index":0,"message":{"role":"assistant","content":"Sure! Here are three numbers between 1 and 10:\n\n1. 5\n2. 8\n3. 9"},"finish_reason":"stop"}],"usage":{"prompt_tokens":51,"total_tokens":81,"completion_tokens":30}} 
+```
+If the container has not been built, it may return that and you need to wait for a while. If it returns like this all the time, there may be something wrong.
```

Comment:
```suggestion
If the container has not been built, it may return this message, and you should wait for a while. If it consistently returns this message, there might be an issue:
```

Reply:
updated